basis/math/matrices/matrices.factor

```diff
@@ -6,25 +6,49 @@ sequences.private fry math.statistics grouping
 combinators.short-circuit math.ranges combinators.smart ;
 IN: math.matrices
 
+DEFER: eye
+<PRIVATE
+: keep-order ( ..a x quot: ( ..a x -- ..b ) -- ..b x )
+    [ dup ] dip call ; inline
```

Comment:
not a big fan because the quotation is called not on ``( ..a x -- )`` but on ``( ..a x x -- )`` since you duplicate the X before calling it?  Also the output is `` -- x ..b )``.  Better maybe to use ``keep swap`` or something.